### ISSUE TITLE
[Feature Request] enhanced email marketing

### ISSUE THREAD

**Issue by g00g1** [FOSSBilling/FOSSBilling#1386]
**Is your feature request related to a problem? Please describe.**
There is Mass Mail Tool extension at FOSSBilling, but capabilities of the Recipients filter are incredibly low - there is even no possibility for filtering clients that opted into email marketing.

**Describe the solution you'd like**
I think two things could be done:

1. add possibility to change custom field type (currently it is string)
2. add optional filter for Mass Mail Tool which checks selected custom field for true/false

**Describe alternatives you've considered**
Currently the only alternative I am aware of is usage of external means, for example something like public read-only mailing list

**Additional context**
null

**Comment by Anuril** [comment]
You're right, there needs to be changes, also to comply with GDPR. I'll look into it and update the module.

**Comment by John-S4** [comment]
Related to #254
